### PR TITLE
Frontend Prototype + Lobby (LAN and Internet)

### PR DESCRIPTION
**Pull request by JayFoxRox**
# DO NOT MERGE THIS

---

This is a prototype for the Qt GUI. It's really hacked together and not functional yet.
I'm playing around with design ideas and try to figure out what we'll need in the lobby / room classes to communicate with the frontend.

(Uploaded so nobody starts working on this from scratch; we'll probably want to rewrite / refactor this A LOT, but it's already a collection of ideas and some useful bits of Qt code)

TODO:
- [ ] Rename RoomDestroyed to RoomNotFound
- [ ] Remove dead code
- [ ] Add handlers for chat messages
- [x] Add routines to add chat or status messages to the log
- [ ] Add some state to handle disconnected / connected mode switches
- [ ] Synchronize the playerlist using a slot for OnRoomInformationChanged.. or something
- [ ] Move to a seperate repo / project / folder w/e
- [ ] Refactor all the code etc.
- [ ] Seperate the sample / unit test code

Follow-ups / issues:
- [ ] Room-Viewer: Blend wifi reception icons + a channel index to symbolize where each person is communicating and how much
- [ ] Room-Viewer: Block sending empty messages by disabling the "Say" button in the GUI if the chat field is empty
- [ ] Room-List: Block connecting with empty nickname or empty server address by disabling Connect button in GUI

### CODE REVIEW

File: src/lobby.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+RakNet::TimeMS quitTime;
+// Holds packets
+RakNet::Packet* p;  
+
+public:
+
+struct RoomInformation {
+  std::string name;
+  //FIXME: Can we somehow extract a game title id from UDS or something?
+  //FIXME: Can we somehow check the NWM cryptokey for validity?
```
Comment (Subv) [inline src/lobby.cpp]: We can compare it against the real key ;)
We could also hash it and compare the hash, i think that's more "legal" in the US?

Comment (JayFoxRox) [reply src/lobby.cpp]: yeah - I was also considering to hash it

File: src/frontend/room_view_window.cpp [hunk 2]
```diff
@@ -350,12 +359,54 @@ void RoomViewWindow::RestoreUIState() {
...
+
+        std::vector<std::string> elements = {
+            member.nickname,
+            member.game_name,
+            MacAddressString(member.mac_address).toStdString(),
+            "- %",
+            "- ms"
+        };
+        for(auto& item : elements) {
+		        QStandardItem *child = new QStandardItem(QString::fromStdString(item));
```
Comment (JayFoxRox) [inline src/frontend/room_view_window.cpp:390]: urgh indentation messed up

**Pull request closed without merging**